File: 0031-next-permutation/0031-next-permutation.py
class Solution:
    def nextPermutation(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        left = len(nums)-2
        right = len(nums)-1
        last = len(nums)-1
        if left == right:
            return
        flag = 0
        while left>=0 and flag == 0:
            if nums[left]>=nums[right]:
                left-=1
                right-=1
            else:
                while (nums[last] <= nums[left]):
                    last-=1
                nums[last],nums[left] = nums[left],nums[last]
                last = len(nums)-1
                nums[left+1:] = sorted(nums[left+1:])
                flag = 1
        if flag == 0:
            nums.sort()

File: 0031-next-permutation/0031-next-permutation.java
import java.util.*;
class Solution {
    public void nextPermutation(int[] nums) {
        int left = nums.length -2;
        int right = nums.length-1;
        int last = nums.length-1;
        if (left == right){
            return;
        }
        int flag = 0;
        while (left>=0 && flag == 0){
            System.out.println(left+" "+right);
            if (nums[left]>=nums[right]){
                left--;
                right--;
            }else {
                while (nums[last]<=nums[left]){
                    last--;
                }
                int temp = nums[left];
                nums[left] = nums[last];
                nums[last] = temp;
                Arrays.sort(nums, left+1, nums.length);
                flag = 1;
            }
        }
        if (flag == 0){
            Arrays.sort(nums);
        }
    }
}